### PR TITLE
Attempt using the latest Boost release with NVCC

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

**Comment by aprokop** [comment]
Yeah, only 1.75 works, 1.74 does not.

**Comment by dalg24** [comment]
Raising a warning instead of an error for versions [1.69, 1.74]

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -36,8 +36,8 @@ find_package(Boost 1.67.0 REQUIRED COMPONENTS unit_test_framework)
 if(NOT DEFINED Boost_VERSION_MINOR)
   set(Boost_VERSION_MINOR ${Boost_MINOR_VERSION})
 endif()
-if(Kokkos_ENABLE_CUDA AND Boost_VERSION_MINOR GREATER_EQUAL 69)
-  message(FATAL_ERROR "Tests in CUDA-enabled ArborX are only compatible with Boost version 1.67 or 1.68")
+if(Kokkos_ENABLE_CUDA AND Boost_VERSION VERSION_GREATER 1.68 AND Boost_VERSION VERSION_LESS 1.75)
+  message(WARNING "Boost versions 1.69 to 1.74 are known to yield build issues with NVCC")
```
[the remark this answers is not in the document]
Comment (dalg24) [reply test/CMakeLists.txt:40]: I believe it works with Clang+CUDA

**Review by masterleinad — APPROVED** [review]

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -36,8 +36,8 @@ find_package(Boost 1.67.0 REQUIRED COMPONENTS unit_test_framework)
 if(NOT DEFINED Boost_VERSION_MINOR)
   set(Boost_VERSION_MINOR ${Boost_MINOR_VERSION})
 endif()
-if(Kokkos_ENABLE_CUDA AND Boost_VERSION_MINOR GREATER_EQUAL 69)
-  message(FATAL_ERROR "Tests in CUDA-enabled ArborX are only compatible with Boost version 1.67 or 1.68")
+if(Kokkos_ENABLE_CUDA AND Boost_VERSION VERSION_GREATER 1.68 AND Boost_VERSION VERSION_LESS 1.75)
```
Comment (aprokop) [inline test/CMakeLists.txt:39]: Note the change from `Boost_VERSION_MINOR` to `Boost_VERSION`. I *think* it should be OK, but could you please confirm? There was some change in its meaning in 3.15 and I'm also not sure Boost versions prior to 1.70 provide it correctly.

Comment (dalg24) [reply test/CMakeLists.txt:39]: I tested locally.

**Review by aprokop — APPROVED** [review]

**Comment by aprokop** [comment]
The test failure is due to warnings from a container rebuild. Ignoring.

**Pull request merged**